contracts/tests/test_unruggable_memecoin.cairo

```diff
@@ -1221,4 +1221,250 @@ mod memecoin_internals {
             index += 1;
         };
     }
+    #[test]
+    fn test_set_merkle_root() {
+        let (
+            owner,
+            recipient,
+            name,
+            symbol,
+            initial_supply,
+            initial_holder_1,
+            initial_holder_2,
+            _,
+            initial_holders_amounts
+        ) =
+            instantiate_params();
+        let alice = contract_address_const::<53>();
+        let initial_holders = array![owner, initial_holder_1, initial_holder_2].span();
+
+        let contract_address =
+            match deploy_contract(
+                owner, owner, name, symbol, initial_supply, initial_holders, initial_holders_amounts
+            ) {
+            Result::Ok(address) => address,
+            Result::Err(msg) => panic(msg.panic_data),
+        };
+
+        let memecoin = IUnruggableMemecoinDispatcher { contract_address };
+
+        start_prank(CheatTarget::One(memecoin.contract_address), owner);
+        memecoin.set_merkle_root(0x3022e5c16bd1bf6e9c44b0d0de23ef6eb0bc84bd2b4eaca75306076eb99239a);
+
+        assert(
+            memecoin
+                .get_merkle_root() == 0x3022e5c16bd1bf6e9c44b0d0de23ef6eb0bc84bd2b4eaca75306076eb99239a,
+            'Inconsistency'
```

Comment:
error message should be descriptive of the error encountered.

Reply:
Got it